### PR TITLE
Access control: Team role picker

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR adds role picker to the _Configuration/Teams_ page.

### CODE REVIEW

File: public/app/core/components/RolePicker/RolePickerInput.tsx [hunk 1]
```diff
@@ -47,8 +49,8 @@ export const RolePickerInput = ({
 
   return !isFocused ? (
     <div className={styles.selectedRoles} onMouseDown={onOpen}>
-      <ValueContainer>{builtInRole}</ValueContainer>
-      {!!numberOfRoles && (
+      {showBuiltInRole && <ValueContainer>{builtInRole}</ValueContainer>}
+      {!!numberOfRoles ? (
```
Comment (hugohaggmark) [inline public/app/core/components/RolePicker/RolePickerInput.tsx:53]: nit: This is getting quite hard to read and reason about. Could this part be simplified somehow? Maybe extracting smaller components ?

File: public/app/core/components/RolePicker/TeamRolePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      } catch (e) {
+        // TODO handle error
+        console.error('Error loading options');
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    fetchOptions();
+  }, [getRoleOptions, orgId, teamId]);
```
Comment (hugohaggmark) [inline public/app/core/components/RolePicker/TeamRolePicker.tsx:36]: Maybe use `useAsync` hook from `react-use` here instead?

File: public/app/core/components/RolePicker/TeamRolePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+export const updateTeamRoles = (roleUids: string[], teamId: number, orgId?: number) => {
+  let teamRolesUrl = `/api/access-control/teams/${teamId}/roles`;
+  if (orgId) {
+    teamRolesUrl += `?targetOrgId=${orgId}`;
+  }
+  return getBackendSrv().put(teamRolesUrl, {
+    orgId,
+    roleUids,
+  });
+};
```
Comment (hugohaggmark) [inline public/app/core/components/RolePicker/TeamRolePicker.tsx:96]: Move this to a separate `api.ts` file. Makes it a lot easier to mock and write tests.

File: public/app/core/components/RolePicker/UserRolePicker.tsx [hunk 4]
```diff
@@ -24,16 +24,45 @@ export const UserRolePicker: FC<Props> = ({
...
+      } catch (e) {
+        // TODO handle error
+        console.error('Error loading options');
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    fetchOptions();
+  }, [getBuiltinRoles, getRoleOptions, orgId, userId]);
```
Comment (hugohaggmark) [inline public/app/core/components/RolePicker/UserRolePicker.tsx:52]: Maybe use useAsync hook from react-use here instead?

**Review by hugohaggmark — COMMENTED** [review]
Code looks good but I found a few things that you might want to consider. Also when running this PR and navigating to the teams page I get the following:
image

Should I be running something before checking out this PR, maybe I'm missing a feature toggle?

**Comment by alexanderzobnin** [comment]
@hugohaggmark Thanks for the feedback, I'll fix issues you mentioned.

File: public/app/core/components/RolePicker/TeamRolePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      let options = await (getRoleOptions ? getRoleOptions() : fetchRoleOptions(orgId));
+      setRoleOptions(options.filter((option) => !option.name?.startsWith('managed:')));
+
+      const teamRoles = await fetchTeamRoles(teamId, orgId);
+      setAppliedRoles(teamRoles);
+    } catch (e) {
+      // TODO handle error
+      console.error('Error loading options');
+    } finally {
+      setIsLoading(false);
```
Comment (hugohaggmark) [inline public/app/core/components/RolePicker/TeamRolePicker.tsx:31]: nit: I think the is loading parameter is exposed from the `useAsync` so you don't need to handle that state.

File: public/app/core/components/RolePicker/UserRolePicker.tsx [hunk 6]
```diff
@@ -24,64 +25,41 @@ export const UserRolePicker: FC<Props> = ({
...
+      const builtInRoles = await (getBuiltinRoles ? getBuiltinRoles() : fetchBuiltinRoles(orgId));
+      setBuiltinRoles(builtInRoles);
+
+      const userRoles = await fetchUserRoles(userId, orgId);
+      setAppliedRoles(userRoles);
+    } catch (e) {
+      // TODO handle error
+      console.error('Error loading options');
+    } finally {
+      setIsLoading(false);
```
Comment (hugohaggmark) [inline public/app/core/components/RolePicker/UserRolePicker.tsx:47]: nit: I think the is loading parameter is exposed from the `useAsync` so you don't need to handle that state.

**Review by hugohaggmark — APPROVED** [review]
Looks great and works as expected 🎉

**Pull request merged**